### PR TITLE
Update zpool module to support new bits

### PR DESCRIPTION
**Pull request by sjorge**
### What does this PR do?
* zpool.scrub now support pausing of a scrub
* zpool.labelclear was added, this allows clearing of zfs labels from devices
* zpool.split was added, this allow splitting of a mirrored zpool
* zpool.create now has support for creating boot partitions.

There is no update to the zpool state, the boot partition can be create using the state by specifying the bootsize property. The choice of not adding a way to create it with the default size is deliberate.

### What issues does this PR fix or reference?
Bring the zpool module inline with recent changes

### New Behavior
Remove this section if not relevant

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by sjorge** [comment]
Seems the test failure is unrelated?

File: salt/modules/zpool.py [hunk 1]
```diff
@@ -537,6 +537,13 @@ def scrub(zpool, stop=False):
         name of storage pool
     stop : boolean
         if true, cancel ongoing scrub
+    pause : boolean
```
Comment (cachedout) [inline salt/modules/zpool.py]: Hrm. Is it better to have this as a kwarg or as an entirely new function? I think I could go either way but it's worth discussing, I think.

Comment (sjorge) [reply salt/modules/zpool.py]: scrub already had a kwarg for stop, so adding one for pause seems logical.

This also fits with the underlying command

```
# start the scrub
zfs scrub mypool
# pause the scrub
zfs scrub -p mypool
# let stop the scrub
zfs scrub -s mypool
```

**Comment by sjorge** [comment]
@cachedout bump

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by sjorge** [comment]
Once this get in I will start trying to fix the issues mentioned in #44404

**Comment by sjorge** [comment]
Yay, looks like Jenkins is now happy, aside from GPG!

**Comment by sjorge** [comment]
@cachedout bump

**Comment by rallytime** [comment]
Hey @sjorge - would you mind writing some unit tests for these new functions?

**Comment by sjorge** [comment]
@rallytime I'll have a look at that this weekend, I might need some extra help with that but I'll see how far I can get.

**Comment by rallytime** [comment]
@sjorge Yeah - let us know if we can help! :0

**Comment by sjorge** [comment]
@rallytime just now had time to look at this, after installing some new dependencies I was missing I got the unit tests to run again locally,... I did notice this:

```Untracked files:
  (use "git add <file>..." to include in what will be committed)

        scripts/suse/zypper/plugins/commit/zyppnotifyc```

Should this file not be in the .gitignore list?

**Comment by sjorge** [comment]
I'm also stuck trying to write a test for split :(

```
    def test_split_success(self):
        '''
        Tests split on success
        '''
        ret = {}
        ret['stdout'] = ""
        ret['stderr'] = ""
        ret['retcode'] = 0
        mock_cmd = MagicMock(return_value=ret)

        @patch('zpool.exists')
        def exists(pool):
            return pool == 'datapool'

        with patch.dict(zpool.__salt__, {'cmd.run_all': mock_cmd}):
            ret = zpool.split('datapool', 'backuppool')
            res = OrderedDict([('backuppool', 'split off from datapool')])
            self.assertEqual(res, ret)
```

zpool.split will call exists() twice, once to check if the original pool exists and once to check if the new pool already exists. So I need to have it return True and False once before the other stuff will work. I can't figure out how to do this.

**Comment by sjorge** [comment]
Got the tests to worked, I swapped out the calls so they no longer call exists() directly but use __salt__['zpool.exists'] this also makes the code a bit more readable because it is now more clear what 'exists' means in those cases.

**Review by rallytime — APPROVED** [review]

**Comment by sjorge** [comment]
Thanks! I’ll try to tackle #44404 this weekend.

**Pull request merged**